packages/server/src/automations/steps/queryRows.js

```diff
@@ -103,6 +119,9 @@ function typeCoercion(filters, table) {
   return filters
 }
 
+const hasNullFilters = filters =>
+  filters.length === 0 || filters.some(filter => filter.value === null)
```

Comment:
The main reason for the current behaviour is that empty strings are the biggest problem - so those are important ones to filter out here. We should update this to `filter.value == null || filter.value === ""` to ensure we account for those.